src/tools/clippy/clippy_lints/src/option_if_let_else.rs

```diff
@@ -86,20 +87,20 @@ struct OptionIfLetElseOccurence {
 
 /// Extracts the body of a given arm. If the arm contains only an expression,
 /// then it returns the expression. Otherwise, it returns the entire block
-fn extract_body_from_arm<'a>(arm: &'a Arm<'a>) -> Option<&'a Expr<'a>> {
+fn extract_body_from_expr<'a>(expr: &'a Expr<'a>) -> Option<&'a Expr<'a>> {
     if let ExprKind::Block(
         Block {
-            stmts: statements,
-            expr: Some(expr),
+            expr: Some(ref block_expr),
+            stmts: ref block_stmts,
```

Comment:
Also here `&expr.kind` over `ref` everywhere. And big NIT: swapping `expr` and `stmts` is unnecessary and makes the code slightly less intuitive (stmts always come before exprs).